frontend/src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service.ts

```diff
@@ -5,76 +5,133 @@
 
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { BehaviorSubject, Observable, tap } from 'rxjs';
-import { ToastService } from 'src/app/helpers/toast/toast.service';
+import {
+  AbstractControl,
+  AsyncValidatorFn,
+  ValidationErrors,
+} from '@angular/forms';
+import { BehaviorSubject, Observable, map, take, tap } from 'rxjs';
 import { T4CInstance } from 'src/app/services/settings/t4c-instance.service';
 import { environment } from 'src/environments/environment';
 
 @Injectable({
   providedIn: 'root',
 })
 export class T4CRepoService {
-  constructor(private http: HttpClient, private toastService: ToastService) {}
+  constructor(private http: HttpClient) {}
 
-  _repositories = new BehaviorSubject<T4CServerRepository[]>([]);
-  get repositories(): T4CServerRepository[] {
-    return this._repositories.value;
-  }
+  private _repositories = new BehaviorSubject<
+    T4CServerRepository[] | undefined
+  >(undefined);
+
+  readonly repositories = this._repositories.asObservable();
 
-  urlFactory(instance_id: number): string {
+  backendURLFactory(instance_id: number): string {
     return `${environment.backend_url}/settings/modelsources/t4c/${instance_id}/repositories/`;
   }
```

Comment:
Why the rename? We used `urlFactory` in other parts of the application.

Reply:
I am not sure why I renamed it like this, so I would revert it. However, when looking at it again I renamed this here to `instanceUrlFactory(...)` and added another `repositoryUrlFactory(...)`